File: algorithms/110. Balanced Binary Tree/solution.cc
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
    bool balanced = true;
public:
    bool isBalanced(TreeNode* root) {
        getDepth(root);
        return balanced;
    }

    int getDepth(TreeNode * root) {
        if ( NULL == root ) {
            return 0;
        }
        int left = getDepth(root->left);
        int right = getDepth(root->right);
        if ( abs(left - right) > 1 ) {
            balanced = false;
        }
        return max(left, right) + 1;
    }
};

File: algorithms/110. Balanced Binary Tree/solution.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def isBalanced(self, root):
        """
        :type root: TreeNode
        :rtype: bool
        """
        def depth(root):
            if not root:
                return 0
            left = depth(root.left)
            right = depth(root.right)
            if abs(left - right) > 1:
                balanced[0] = False
            return max(left, right) + 1
        balanced = [True]
        depth(root)
        return balanced[0]
